pints/plot/_surface.py

```diff
@@ -0,0 +1,324 @@
+    voronoi region around each point within the boundaries specified by
+    ``xlim`` and ``ylim``.
+
+    Points and voronoi regions entirely outside the specified bounds will be
+    dropped. Voronoi regions partially outside the bounds will be truncated.
+    The third array ``f`` will be filtered the same way as ``x`` and ``y`` but
+    is otherwise not used.
+
+    Parameters
+    ----------
+    x
+        A list of x-coorindates
+    y
+        A list of y-coordinates
+    f
+        The score function at the given x and y coordinates
+    xlim
+        Lower and upper bound for the x coordinates
+    ylim
+        Lower and upper bound for the y coordinates
+
+    Returns
+    -------
+    A tuple ``(x, y, f, regions)`` where ``x``, ``y`` and ``f`` are the
+    coordinates of the accepted points and each ``regions[i]`` is a list of the
+    vertices making up the voronoi region for point ``(x[i], y[i])``.
+    """
+    from scipy.spatial import Voronoi
+    try:
+        from itertools import izip  # Python 2's izip acts like Python 3's zip
+    except ImportError:
+        izip = zip
+
+    # Don't check x, y, f: handled by calling method
+    n = len(x)
+
+    # Check limits
+    xmin, xmax = [float(a) for a in sorted(xlim)]
+    ymin, ymax = [float(a) for a in sorted(ylim)]
+
+    # Drop any points outside the bounds
+    # within_bounds = (x >= xmin) & (x <= xmax) & (y >= ymin) & (y <= ymax)
+    # x = x[within_bounds]
+    # y = y[within_bounds]
+    # f = f[within_bounds]
+
+    # Create voronoi diagram
+    vor = Voronoi(np.array([x, y]).transpose())
+
+    # The voronoi diagram consists of a number of ridges drawn between a set
+    # of points.
+    #
+    #   points          Are the points the diagram is based on.
+    #   vertices        The coordinates of the vertices connecting the ridges
+    #   ridge_points    Is a list of tuples (p1, p2) defining the points each
+    #                   ridge belongs to. Points are given as their index in
+    #                   the list of points.
+    #   ridge_vertices  Is a list of vertices (v1, v2) defining the vertices
+    #                   between which each ridge is drawn. Vertices are given
+    #                   as their index in the list of vertice coordinates. For
+    #                   ridges extending to infinity, one of the vertices will
+    #                   be given as -1.
+    #
+    # Get the center of the voronoi diagram's points and define a radius that
+    # will bring us outside the visible area for any starting point / direction
+    #
+    center = vor.points.mean(axis=0)
+    radius2 = 2 * np.sqrt((xmax - xmin)**2 + (ymax - ymin)**2)
+
+    # Create a list containing the set of vertices defining each region
+    regions = [set() for i in range(n)]
+    for (p1, p2), (v1, v2) in izip(vor.ridge_points, vor.ridge_vertices):
+        # Ensure only v1 can every be -1
+        v1, v2 = (v2, v1) if v1 > v2 else (v1, v2)
```

Comment:
what does the comment have to do with v1 < v2?

Reply:
it's probably just me not knowing how voronoi works

Reply:
It's kind of explained in line 161-165. v1, v2 are non-negative integers except the inf one being -1, so that if v1 > v2 then swap them makes sure -1 must go first.

Obviously I agree that the line does more than what the comment wants!